dot/rpc/modules/grandpa_integration_test.go

```diff
@@ -0,0 +1,114 @@
+// Copyright 2021 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+//go:build integration
+// +build integration
+
+package modules
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/ChainSafe/gossamer/dot/state"
+	"github.com/ChainSafe/gossamer/dot/types"
+	"github.com/ChainSafe/gossamer/lib/crypto/ed25519"
+	"github.com/ChainSafe/gossamer/lib/grandpa"
+	"github.com/ChainSafe/gossamer/lib/keystore"
+	"github.com/stretchr/testify/require"
+
+	rpcmocks "github.com/ChainSafe/gossamer/dot/rpc/modules/mocks"
+)
+
+var kr, _ = keystore.NewEd25519Keyring()
+
+func TestGrandpaProveFinality(t *testing.T) {
+	testStateService := newTestStateService(t)
+
+	state.AddBlocksToState(t, testStateService.Block, 3, false)
+	bestBlock, err := testStateService.Block.BestBlock()
+
+	if err != nil {
+		t.Errorf("Fail: bestblock failed")
+	}
+
+	gmSvc := NewGrandpaModule(testStateService.Block, nil)
+
+	testStateService.Block.SetJustification(bestBlock.Header.ParentHash, make([]byte, 10))
+	testStateService.Block.SetJustification(bestBlock.Header.Hash(), make([]byte, 11))
+
+	var expectedResponse ProveFinalityResponse
+	expectedResponse = append(expectedResponse, make([]byte, 10), make([]byte, 11))
+
+	res := new(ProveFinalityResponse)
+	err = gmSvc.ProveFinality(nil, &ProveFinalityRequest{
+		blockHashStart: bestBlock.Header.ParentHash,
+		blockHashEnd:   bestBlock.Header.Hash(),
+	}, res)
+
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if !reflect.DeepEqual(*res, expectedResponse) {
+		t.Errorf("Fail: expected: %+v got: %+v\n", res, &expectedResponse)
+	}
```

Comment:
You're probably just moving tests around, sorry for the suggestion noise! 

```suggestion
	assert.Equal(t, expectedResponse, *res)
```